sdk/servicebus/azure-servicebus/azure/servicebus/_common/_configuration.py

```diff
@@ -23,3 +23,4 @@ def __init__(self, **kwargs):
         self.auth_timeout = kwargs.get("auth_timeout", 60)  # type: int
         self.encoding = kwargs.get("encoding", "UTF-8")
         self.auto_reconnect = kwargs.get("auto_reconnect", True)
+        self.keep_alive = kwargs.get("keep_alive", 30)
```

Comment:
I assume 30 aligns with cross-language consistency? (just double-checking)

Reply:
This 30 seconds default value is used in eventhub track1.

The working mechanism in uamqp is "every 30 seconds, call connection.do_work(), connection.do_work() would check whether it has passed 0.5*remote idle timeout (240s), if passed, send a empty frame out"

.Net is sending out empty frame like every ~ 50s seconds.
JS is sending out empty frame every 0.5 *remote idle timeout.

I think every 30 seconds is a reasonable interval in our case.

Reply:
I'm wondering how we tell the users to configure this value. 240s is the hard expiry time. So is 220s always better than 200s to keep the connection alive because it does the work with less traffic? If so, a bool value is better than a number.

Reply:
@YijunXieMS , right now we don't expose this parameter to users to keep consistent with other languages -- JS and .Net don't allow users to set/tweak this interval, it is turned on by default in their SDK.

If there're customer needs to configure the value/turn on the switch, this could be a post ga feature.

Reply:
This echoes a question I posed for autolockrenew, and I don't recall if we settled on an answer: Should we disallow setting a keepalive > 240s?  Or caveat emptor?  I might classify it a "semantic error"; if someone wants to disable keepalive they should show intent and pass None.  (Context: Have had at least one user who adjusted a value such as this and unintentionally ran into lock expiry as a result)

Reply:
as person discussion OOB:

The proposal of flag is preferred as it's simple, users don't need to care about the value.
But we will leave it untouched (turned on by default) until there're customer requests for turning it off.